lib/charms/tls_certificates_interface/v0/tls_certificates.py

```diff
@@ -458,13 +458,12 @@ def _on_relation_changed(self, event) -> None:
         Returns:
             None
         """
-        if self.model.unit.is_leader():
-            relation_data = _load_relation_data(event.relation.data[event.unit])
-            if not self._relation_data_is_valid(relation_data):
-                logger.warning("Relation data did not pass JSON Schema validation - Deferring")
-                event.defer()
-                return
+        relation_data = _load_relation_data(event.relation.data[event.unit])
+        if not self._relation_data_is_valid(relation_data):
```

Comment:
Should we check for `if not relation_data` like in the `_on_relation_changed` method for the `TLSCertificatesProvides` class?

Reply:
Fixed